docs/specs/devcontainer-reference.md

```diff
@@ -0,0 +1,144 @@
+
+There are multiple properties that allow users to control `docker build` works:
+
+- `build.context` 
+- `build.args`
+- `build.target`,
+- `build.cacheFrom`
+
+## Docker Compose based
+
+Docker compose configurations utilize `docker-compose`  to create and manage a set of containers required for an application. As with the other configurations, any images required for this operation are assumed to be reachable. The required parameters are:
+
+- `build.dockerComposeFile`  the reference to the docker-compose file(s) to be used.
+- `service`  this tag declares the **main** container that will be used for all other operations. Tools are assumed to also use this parameter to connect to the **development container** although they can provide facilities to connect to the other containers as required by the user.
+
+It is important to remark that due to how `docker-compose` works a lot of the properties that apply to **image** and **dockerfile** configurations can’t be set by the application and thus must be set up manually by the user using the `docker-compose` format itself.
+
+-`runServices` are the set of services in the `docker-compose` configuration that should be started or stopped with the environment.
+
+# Other options
+
+In addition to the configuration options explained above there are other settings that apply when creating the **development containers** to facilitate their use by developers. These properties mostly apply to **image** and **dockerfile** based configurations.
+
+The main ones are:
+
+## Environment Variables
+
+Environment variables can be set in containers at creation time, to override existing ones, or just to be used when a supporting tool connects to the environment.
+
+There are two kinds of environment variables:
+Container. These variables are part of the container when it is created and are available at all points in its lifecycle.
+Remote. This variables are to be set by a **development container** supporting tool as part of its environment when it runs. This variables can change during the lifetime of the container.
+
+## Mounts
+
+Mounts allow containers to have access to the underlying machine, share data between containers and to persist information between **development containers**. 
+
+It is important to note that these mounts are from the underlying compute environment and thus in cloud environments might not have access to the same data as a local machine.
+
+## Users
+
+Users control the permissions of applications executed in the containers, allowing the developer to control them. The specification takes into account two types of user definuitions:
+
+Container User. The user that will be used for all operations that run inside a container. It is used to run lifecycle commands among others.
+Remote User. In case the developer wants to use different users for different purposes, the remote user is the user that connecting tools should use to execute operations inside the container.
```

Comment:
```suggestion
* Container User: The user that will be used for all operations that run inside a container. It is used to run lifecycle commands among others.
* Remote User: In case the developer wants to use different users for different purposes, the remote user is the user that connecting tools should use to execute operations inside the container.
```